cart/infrastructure/defaultCartBehaviour.go

```diff
@@ -54,21 +57,38 @@ type (
 
 	// DefaultVoucherHandler implements a basic voucher handler
 	DefaultVoucherHandler struct{}
+
+	CartNotFoundError struct {
+		action string
+		id     string
+	}
 )
 
+func (e *CartNotFoundError) Error() string {
```

Comment:
would prefer to have a regular error like:
```
var ErrCartNotFound = errors.New("cart not found")
```

and wrapping it then in places like `fmt.Errorf("%w for cart id %q, during action", err, cart.ID, action)`